client/src/main/java/com/scalar/dl/client/service/ClientService.java

```diff
@@ -87,6 +89,48 @@ public ClientService(
     this.signer = signer;
   }
 
+  /**
+   * Bootstraps the ledger by registering the identity (certificate or secret key) and system
+   * contracts if necessary, based on {@code ClientConfig}. The authentication method (digital
+   * signature or HMAC) is determined by the configuration. If the identity or contract is already
+   * registered, it is simply skipped without throwing an exception.
+   *
+   * @throws ClientException if a request fails for some reason
+   */
+  public void bootstrap() {
+    try {
+      if (config.getAuthenticationMethod().equals(AuthenticationMethod.DIGITAL_SIGNATURE)) {
+        registerCertificate();
+      } else {
+        registerSecret();
+      }
+    } catch (ClientException e) {
+      if (!e.getStatusCode().equals(StatusCode.CERTIFICATE_ALREADY_REGISTERED)
+          && !e.getStatusCode().equals(StatusCode.SECRET_ALREADY_REGISTERED)) {
+        throw e;
+      }
+    }
+
+    registerValidateLedgerContract();
+  }
+
+  private void registerValidateLedgerContract() {
+    if (config.isAuditorEnabled()) {
+      Class<?> clazz = ValidateLedger.class;
+      try {
+        registerContract(
+            config.getAuditorLinearizableValidationContractId(),
```

Comment:
For this version (the first version that bundles the `ValidateLedger` contract), we register the validation contract with the same fixed contract ID that we have used before, for backward compatibility.